### PR TITLE
22072 Show previous correspondences + cleanup

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* bcgov/entity#22072

*Description of changes:*

Commit 1:
- app version = 2.6.51
- imported vue-auto-resize directive for textarea
- added typing to ContinuationApplicationTable.vue
- separated ContinuationAuthorizationReviewResult.vue into 2 new sub-components
- refactored continuation review object into separate review and filing objects/props
- now use for loop for previous correspondence
- use label for correspondences header
- use textarea for correspondence comment
- compute correspondence items according to review/results
- updated Review Result dropdown items
- reset validation when new Review Result is selected
- renamed ContinuationReviewStatus -> ReviewStatus
- changed ACCEPTED to APPROVED
- updated interfaces to match API responses
- added fetchFiling()
- updated mocked API responses
- added dateToPacificTime() and apiToPacificDateTime()
- renamed misc things
- swapped main review section icons per Andy
- refactored data fetching
- added return to staff dashboard on error dialog close
- added/updated unit tests

Commit 2:
- refactored date conversions to use Moment
- deleted date-utils.ts

Commit 3:
- transformed ContinuationAuthorizationReview.vue to Composition API
- transformed ExtraprovincialRegistrationBc.vue to Composition API
- transformed HomeJurisdictionInformation.vue to Composition API
- transformed PreviousCorrespondence.vue to Composition API
- transformed ReviewResult.vue to Composition API
- misc cleanup

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

**Comment by severinbeauvais** [comment]
@pwei1018 I ran "gcb run" in this PR, and I saw the "account-ui-ci-pr" job run, but no comment was posted with the temporary URL. Can I find it somewhere?

**Comment by severinbeauvais** [comment]
## Screenshots

Review id 1:
image

Review id 2:
image

**Comment by severinbeauvais** [comment]
## Screenshots for review id=2

image

image

**Review by ArwenQin — APPROVED** [review]
I reviewed the codes, make sense to me!

**Review by PaulGarewal — APPROVED** [review]
LGTM 👍

**Comment by severinbeauvais** [comment]
@pwei1018 I ran "gcb run" in this PR several times, and I saw the "account-ui-ci-pr" job run, but no comment was posted with the temporary URL. Can I find it somewhere?

**Comment by severinbeauvais** [comment]
@JohnamLane I have 2 approvals but merging is blocked. What's the process to merge this PR?

image

File: auth-web/src/services/business.services.ts
Comment (severinbeauvais) [inline auth-web/src/services/business.services.ts:205]: @ArwenQin  FYI ^^

**Review by seeker25 — CHANGES_REQUESTED** [review]
Use composition-api and moment, not your own rolled date library.

We spent a whole bunch of time converting components to `composition-api`, you're undoing that process by writing them in `class-api`. 

You're also introducing more tech debt by using your own date library, when we are using moment in auth-web.

Karim has written his components in `composition-api`, it was a requirement before merging.

**Comment by severinbeauvais** [comment]
> Use composition-api and moment, not your own rolled date library.
> 
> We spent a whole bunch of time converting components to `composition-api`, you're undoing that process by writing them in `class-api`.
> 
> You're also introducing more tech debt by using your own date library, when we are using moment in auth-web.
> 
> Karim has written his components in `composition-api`, it was a requirement before merging.

I would like Composition API a lot more if it properly supported typing, but:

image

and then

image

It is my understanding that Composition API is not required for Vue3 (or Nuxt), so why change and take a hit along the way?

As for the date-time manipulations, this may be our own rolled up library, but it's been in use since 2019, and I'm going to dig in my heels a bit on Moment because what I've researched says to no longer use it. In fact, using Moment seems to be incurring tech debt. If we are dead-set against using native JS for date-time manipulations then we should look for a modern, light alternative to Moment.

Secondly, since our code is temporarily hosted in Auth Web, I want to make it easy to port out to another UI, so reusing the same date-time methods as the other Entity UIs seems like a good way to keep things consistent.

I agree with you on the goal of modern, simple code, but I think changing to Composition API and Moment are not a step ahead right now. If your research proves me wrong, please let me know. I am making my decisions based on what I know but I'm happy to learn more and come up with better solutions.

**Comment by seeker25** [comment]
The requirements aren't up for discussion, unfortunately - composition-api and moment or this isn't getting merged

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://bcregistry-account-dev--pr-2907-cwo0803q.web.app

**Comment by severinbeauvais** [comment]
> The requirements aren't up for discussion, unfortunately - composition-api and moment (consistent with the rest of the repo) or this isn't getting merged

Please provide a reference to the requirements. Thank you.

**Comment by seeker25** [comment]
Here they are: Composition-api and moment

**Comment by severinbeauvais** [comment]
@seeker25 So this is the library that you want me to use for date/time manipulations, even though it's a legacy project, now in maintenance mode, and they say we should choose a different library?

image

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://bcregistry-account-dev--pr-2907-cwo0803q.web.app

**Comment by seeker25** [comment]
Yeah we're using Luxon for Vue3 FAS-UI

File: auth-web/src/components/auth/staff/continuation-application/ExtraprovincialRegistrationBc.vue
Comment (severinbeauvais) [inline auth-web/src/components/auth/staff/continuation-application/ExtraprovincialRegistrationBc.vue:92]: I found that I could use this "getter" syntax instead of "computed", as long as they reference other state property using "this" (eg, line 95).

Travis, are you OK with this?

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://bcregistry-account-dev--pr-2907-cwo0803q.web.app

File: auth-web/src/components/auth/staff/continuation-application/ExtraprovincialRegistrationBc.vue
Comment (severinbeauvais) [inline auth-web/src/components/auth/staff/continuation-application/ExtraprovincialRegistrationBc.vue:115]: This common utils method generates the following warning:

image

Re:

image

I believe it's the locale string that it doesn't like, and that comment about not working in CI is suspicious, too.

However, I tested that all my date-time conversions are working as expected, so 🤷‍♂️

File: auth-web/src/components/auth/staff/continuation-application/HomeJurisdictionInformation.vue
Comment (severinbeauvais) [inline auth-web/src/components/auth/staff/continuation-application/HomeJurisdictionInformation.vue:26]: Interesting that this doesn't need to be `errorDialogComponent.value.close()` (as it does in the script below). Oh well, this works.

File: auth-web/src/components/auth/staff/continuation-application/HomeJurisdictionInformation.vue
Comment (severinbeauvais) [inline auth-web/src/components/auth/staff/continuation-application/HomeJurisdictionInformation.vue:349]: Travis, I did this to circumvent the error "property 'open' does not exist on type 'unknown'".

I assume/hope that this will work out in Vue 3. What do you think?

File: auth-web/src/views/auth/staff/ContinuationAuthorizationReview.vue
Comment (severinbeauvais) [inline auth-web/src/views/auth/staff/ContinuationAuthorizationReview.vue:253]: Travis, I wasn't able to replicate this in Composition API. I googled for example and didn't find any, and '@vue/composition-api' didn't provide a similar lifecycle hook.

How would you deal with this?

**Comment by severinbeauvais** [comment]
I'll fix the unit test errors tomorrow!

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://bcregistry-account-dev--pr-2907-cwo0803q.web.app

File: auth-web/src/components/auth/staff/continuation-application/ExtraprovincialRegistrationBc.vue
Comment (seeker25) [reply auth-web/src/components/auth/staff/continuation-application/ExtraprovincialRegistrationBc.vue:92]: Shouldn't be using `this` composition-api was trying to get away from the `this` context

File: auth-web/src/components/auth/staff/continuation-application/HomeJurisdictionInformation.vue
Comment (seeker25) [reply auth-web/src/components/auth/staff/continuation-application/HomeJurisdictionInformation.vue:349]: will be fine for Vue3, just typing

File: auth-web/src/views/auth/staff/ContinuationAuthorizationReview.vue
Comment (seeker25) [reply auth-web/src/views/auth/staff/ContinuationAuthorizationReview.vue:253]: image

Comment (seeker25) [reply auth-web/src/views/auth/staff/ContinuationAuthorizationReview.vue:253]: https://router.vuejs.org/guide/advanced/composition-api.html#navigation-guards

File: auth-web/src/components/auth/staff/continuation-application/ExtraprovincialRegistrationBc.vue
Comment (seeker25) [reply auth-web/src/components/auth/staff/continuation-application/ExtraprovincialRegistrationBc.vue:115]: Would just use moment tz https://momentjs.com/timezone/

`moment(date).tz('America/Vancouver')`

File: auth-web/src/components/auth/staff/continuation-application/ExtraprovincialRegistrationBc.vue
Comment (severinbeauvais) [reply auth-web/src/components/auth/staff/continuation-application/ExtraprovincialRegistrationBc.vue:92]: OK. Fixing it now.

File: auth-web/src/views/auth/staff/ContinuationAuthorizationReview.vue
Comment (severinbeauvais) [reply auth-web/src/views/auth/staff/ContinuationAuthorizationReview.vue:253]: Thanks. Looking...

File: auth-web/src/components/auth/staff/continuation-application/ExtraprovincialRegistrationBc.vue
Comment (severinbeauvais) [reply auth-web/src/components/auth/staff/continuation-application/ExtraprovincialRegistrationBc.vue]: Fixed in latest commit. Resolving

File: auth-web/src/components/auth/staff/continuation-application/ExtraprovincialRegistrationBc.vue
Comment (severinbeauvais) [reply auth-web/src/components/auth/staff/continuation-application/ExtraprovincialRegistrationBc.vue]: OK, `moment.tz()` works without producing a warning but I needed to import "moment-timezone" library (another 3MB).

What do you think?

File: auth-web/src/views/auth/staff/ContinuationAuthorizationReview.vue
Comment (severinbeauvais) [reply auth-web/src/views/auth/staff/ContinuationAuthorizationReview.vue:253]: The example above didn't work in my component, but I came up with something that works as desired. Please have a look. Resolving.

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://bcregistry-account-dev--pr-2907-cwo0803q.web.app

File: auth-web/tests/unit/components/ContinuationAuthorizationReviewResult.spec.ts
Comment (severinbeauvais) [inline auth-web/tests/unit/components/ContinuationAuthorizationReviewResult.spec.ts:1]: image

File: auth-web/tests/unit/components/ExtraprovincialRegistrationBc.spec.ts
Comment (severinbeauvais) [inline auth-web/tests/unit/components/ExtraprovincialRegistrationBc.spec.ts:1]: image

File: auth-web/tests/unit/components/HomeJurisdictionInformation.spec.ts
Comment (severinbeauvais) [inline auth-web/tests/unit/components/HomeJurisdictionInformation.spec.ts:1]: image

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://bcregistry-account-dev--pr-2907-cwo0803q.web.app

File: auth-web/tests/unit/components/PreviousCorrespondence.spec.ts
Comment (severinbeauvais) [inline auth-web/tests/unit/components/PreviousCorrespondence.spec.ts:1]: image

File: auth-web/tests/unit/components/ReviewResult.spec.ts
Comment (severinbeauvais) [inline auth-web/tests/unit/components/ReviewResult.spec.ts:1]: image

File: auth-web/tests/unit/views/ContinuationAuthorizationReview.spec.ts
Comment (severinbeauvais) [inline auth-web/tests/unit/views/ContinuationAuthorizationReview.spec.ts:1]: image

**Comment by severinbeauvais** [comment]
@seeker25 Ready for re-review and hopefully, merging :)

File: auth-web/src/components/auth/staff/continuation-application/ExtraprovincialRegistrationBc.vue
Comment (seeker25) [reply auth-web/src/components/auth/staff/continuation-application/ExtraprovincialRegistrationBc.vue]: 3MB is the unpacked size

it's fine, probably will replace it all with Luxon anyways

Comment (seeker25) [reply auth-web/src/components/auth/staff/continuation-application/ExtraprovincialRegistrationBc.vue]: image

File: auth-web/src/components/auth/staff/continuation-application/HomeJurisdictionInformation.vue
Comment (seeker25) [inline auth-web/src/components/auth/staff/continuation-application/HomeJurisdictionInformation.vue:367]: `...toRefs(state)` here, above `...state` should they be the same?

File: auth-web/src/components/auth/staff/continuation-application/ExtraprovincialRegistrationBc.vue
Comment (seeker25) [inline auth-web/src/components/auth/staff/continuation-application/ExtraprovincialRegistrationBc.vue:118]: ...toRefs(state)?

File: auth-web/src/components/auth/staff/continuation-application/ContinuationApplications.vue
Comment (severinbeauvais) [reply auth-web/src/components/auth/staff/continuation-application/ContinuationApplications.vue:26]: Arwen worked on this component. I'll ask her to transform it to Composition API.

Comment (severinbeauvais) [reply auth-web/src/components/auth/staff/continuation-application/ContinuationApplications.vue:26]: @ArwenQin ^^

File: auth-web/src/components/auth/staff/continuation-application/PreviousCorrespondence.vue
Comment (seeker25) [inline auth-web/src/components/auth/staff/continuation-application/PreviousCorrespondence.vue:139]: ...toRefs(state) ? not sure if it makes a diff

File: auth-web/src/views/auth/staff/ContinuationAuthorizationReview.vue
Comment (seeker25) [inline auth-web/src/views/auth/staff/ContinuationAuthorizationReview.vue:181]: `state.haveUnsavedChanges`

File: auth-web/src/components/auth/staff/continuation-application/HomeJurisdictionInformation.vue
Comment (severinbeauvais) [reply auth-web/src/components/auth/staff/continuation-application/HomeJurisdictionInformation.vue:367]: It's my understanding that `toRefs()` needs to be used because I declare a ref on line 256.

I will try `...state` and see if it still works.

File: auth-web/src/components/auth/staff/continuation-application/ExtraprovincialRegistrationBc.vue
Comment (severinbeauvais) [reply auth-web/src/components/auth/staff/continuation-application/ExtraprovincialRegistrationBc.vue:118]: No, because I don't declare refs in this component.

File: auth-web/src/components/auth/staff/continuation-application/PreviousCorrespondence.vue
Comment (severinbeauvais) [reply auth-web/src/components/auth/staff/continuation-application/PreviousCorrespondence.vue:139]: No refs declared in this component.

**Review by seeker25 — APPROVED** [review]
LGTM

File: auth-web/src/components/auth/staff/continuation-application/HomeJurisdictionInformation.vue
Comment (seeker25) [reply auth-web/src/components/auth/staff/continuation-application/HomeJurisdictionInformation.vue:367]: all good, either way works for me

File: auth-web/src/components/auth/staff/continuation-application/PreviousCorrespondence.vue
Comment (seeker25) [reply auth-web/src/components/auth/staff/continuation-application/PreviousCorrespondence.vue:139]: all good, either way works for me

File: auth-web/src/components/auth/staff/continuation-application/ContinuationApplications.vue
Comment (ArwenQin) [reply auth-web/src/components/auth/staff/continuation-application/ContinuationApplications.vue:26]: Got it. Can I do it in ticket #21983 (the enhanced table ticket I'm working on)?

Comment (seeker25) [reply auth-web/src/components/auth/staff/continuation-application/ContinuationApplications.vue:26]: It's a pretty simple one, lets just merge I caused you enough side tracking

File: auth-web/src/components/auth/staff/continuation-application/HomeJurisdictionInformation.vue
Comment (severinbeauvais) [reply auth-web/src/components/auth/staff/continuation-application/HomeJurisdictionInformation.vue:367]: Looks like state isn't correctly injected into the component if I don't use `toRefs()` -- the modal looks like this (missing its props):

image

**Pull request merged**
